docs/dev/howtoTestRepository.md

```diff
@@ -0,0 +1,156 @@
+First of all you should find a Class to test, where you know the logic behind.
+If you follow this example, you will test the Response generated by the backend against a local File you created, it will contain the Response we are expecting.
+
+### Download the Winery Test Repository
+Based on the Winery test Repository the Responses will be created from the backend. Its available under [Github/Winery/test-repository](https://github.com/winery/test-repository)
+
+#### Windows
+
+You will have to check out the Winery Repository to `C:\Winery-Repository`
+That means you will navigate to your C-Directory with an Admin-CMD or Powershell. After that you clone `https://github.com/winery/test-repository.git`
+
+`git clone https://github.com/winery/test-repository.git C:\winery-repository`
+
+Now we have the last stand of the Winery Repository test Data.
+
+#### Unix,Mac,Ubuntu etc.
+
+You will have to check out the Winery Repository to `C:\Winery-Repository`
+That means you will navigate to your Home-Directory with a terminal or your favourite shell. After that you clone `https://github.com/winery/test-repository.git`
+
+`git clone https://github.com/winery/test-repository.git ~/winery-repository`
+
+Now we have the last stand of the Winery Repository test Data.
+
+### How to get the Right Commit ID for your Test
+If you write a new Test you should always base it on the last commit, if you add Files use that commit.
+If you want to know which is the latest commit id right now you should get the SHA1 Commit Id.
+
+To get it from your checkout just follow these Steps:
+
+ - Go to the Github Side [Commit overview](https://github.com/winery/test-repository/commits/black)
+ - Choose your commit and look at the right side - you will see a short code like `88e5ccd` 
+ - Click on the Commit code you want to see
+ - On the Right Side you will see the full Commit ID, it starts with the same as the link you clicked before, in this case `88e5ccd6c35aeffdebc19c8dda9cd76f432538f8`.
+
+ 
+ 
+## Create a new Test
+In this chapter we will cover the notation and the Paths you should place the Test Classes you create.
+
+### Create a Test Class
+If you want to Test the Class `NodeTypeResource` in the Package `org.eclpse.winery.repository.resources.entitytypes.nodetypes`.
+That means we need to create a respective package in the Test Folder of the Project.
+
+We create a new Class for the `NodeTypeResource` Test which we name  `NodeTypeResourceTest` in the Package  `org.eclipse.winery.repository.resources.entitytypes.nodetypes` in the Test Folder located under src/test`
+
+Now that we are Testing a Resource Class we have to add `extends AbstractResourceTest` to the Class definition, this will help us simplifying the Test Classes and only put the tests them self in there.
+
+### Junit-Annotations
+In this Chapter, all Annotations used or needed at time of writing in the existing Tests.
+
+To create Tests its neccessary to use atleast once the `@Test` annotation.
```

Comment:
I would delete all these subsections, because this is common JUnit knowledge and can be read elsewhere.